src/lib/web.py

```diff
@@ -0,0 +1,349 @@
+        workspace_status = str(workspace_status_ele.text).lower()
+        final_workspace_status = ""
+        if workspace_status == "stopped":
+            self.driver.find_element_by_xpath("//button[contains(text(), 'Run')]").click()
+        elif workspace_status == "running":
+            final_workspace_status = "running"
+        cutoff = time.time() + 150
+        while workspace_status == "stopped" or workspace_status == "starting":
+            status_ele = self.driver.find_element_by_xpath("//span[@id='workspace-status']")
+            current_status = str(status_ele.text).lower()
+            if current_status == "running":
+                final_workspace_status = current_status
+                break
+            if time.time() > cutoff:
+                break
+            time.sleep(2)
+        if final_workspace_status != "running":
+            raise Exception("The workspace could not be started !")
+        self.wait_click_element(By.XPATH, "//a[contains(text(), 'Open')]")
+        # Check workspace has loaded
+        self.wait_switch_frame(By.ID, "ide-application-iframe", 90)
+
+    def run_analysis(self):
+        """
+        Runs analysis on project in eclipse che workspace and returns story points
+
+        Args:
+            None
+
+        Returns:
+            story_points (str): Story points for project analysis
+        """
+        # Click on MTA Explorer icon
+        mta_plugin_ele = self.wait_find_element(
+            By.XPATH,
+            "//div/ul[@class='p-TabBar-content']\
+            /li[@id='shell-tab-plugin-view-container:rhamt-explorer']",
+            120,
+        )
+        mta_icon_click_counter = 2
+        if "p-mod-current" not in mta_plugin_ele.get_attribute("class"):
+            mta_icon_click_counter = 3
+        for _ in range(mta_icon_click_counter):
+            mta_plugin_ele.click()
+        # Create new run configuration
+        self.wait_click_element(By.XPATH, "//div[@title='New Configuration']")
+        self.driver.switch_to.default_content()
+        self.wait_switch_frame(By.ID, "ide-application-iframe")
+        # Wait for all the page frames to load
+        time.sleep(30)
+        # Switch to the last frame to find main html content for configuration
+        self.wait_switch_frame(By.TAG_NAME, "iframe")
+        self.wait_switch_frame(By.XPATH, "//iframe[@id='active-frame']", 60)
+        self.wait_switch_frame(By.TAG_NAME, "iframe")
+        # Enter project name and select quarkus 1 as target
+        self.wait_click_element(By.XPATH, "//span[@id='input-details']/dl/dd/div/a")
+        self.write_text(
+            By.XPATH, "//input[@id='editDialogInput']", config_data["workspace_project_path"],
+        )
+        self.driver.find_element_by_xpath("//input[@id='editDialogInput']").send_keys(Keys.ENTER)
+        self.wait_click_element(By.XPATH, "//input[@id='target-quarkus1']")
```

Comment:
Can we pass target as parameter to the method as this can change

Reply:
Will update